src/censure.jl

```diff
@@ -0,0 +1,236 @@
+abstract BiFilter
+
+type BoxFilter <: BiFilter
+
+	scale :: Integer
+	in_length :: Integer
```

Comment:
This is going to hurt performance because `Integer` is an abstract type. Better to make it an `Int` or, if you need flexibility about the type, make `BoxFilter` parametric, e.g.,
```jl
type BoxFilter{T<:Integer} <: BiFilter
    scale::T
    out_length::T
    in_area::Float64
...
end
```